app/controllers/admin_controller.rb

```diff
@@ -11,10 +11,28 @@ def index
     messages = Message.all
 
     total_words = messages.reduce(0) { |sum, message| sum + message.content.split.count }.to_f
-    @average_message_length = total_words / messages.length
+    if messages.length > 0
+      @average_message_length = total_words / messages.length
+    else
+      @average_message_length = 0
+    end
 
     @messages_logged = messages.count
+    @messages_authorized = messages.find_all { |message| message.authorized }.count
+    @messages_sent = messages.find_all { |message| message.sent }.count
 
     @distinct_zips = Message.select(:zip).distinct.length
   end
+
+  def send_messages
+    unsent_messages = Message.where({sent: false, authorized: true})
```

Comment:
what did `authorized` mean again?

Reply:
That they clicked through on the confirmation email.